File: CompassReports.Resources/Models/EnrollmentFilterModel.cs
using System.Collections.Generic;

namespace CompassReports.Resources.Models
{
    public class EnrollmentFilterModel
    {
        public List<int> Districts { get; set; }
        public List<string> EnglishLanguageLearnerStatuses { get; set; }
        public List<string> Ethnicities { get; set; }
        public List<string> Grades { get; set; }
        public List<string> LunchStatuses { get; set; }
        public List<string> SpecialEducationStatuses { get; set; }
        public List<int> Schools { get; set; }
        public short SchoolYear { get; set; }
        public List<short> SchoolYears { get; set; }
    }
}

File: CompassReports.Web/App/Models/enrollment-filter.model.ts
module App.Models {
    export class EnrollmentFilterModel implements IReportFilterModel {

        DefaultSchoolYear: number;
        SchoolYear?: number;
        SchoolYears?: number[];

        EnglishLanguageLearnerStatuses: string[];
        Ethnicities: string[];
        Grades: string[];
        LunchStatuses: string[];
        SpecialEducationStatuses: string[];

        Schools: number[];
        Districts: number[];

        filteringCount = () => {
            //Starts at one since Grades always shows
            let count = 1;

            if (this.EnglishLanguageLearnerStatuses != null && this.EnglishLanguageLearnerStatuses.length) count++;
            if (this.Ethnicities != null && this.Ethnicities.length) count++;
            if (this.SchoolYears != null && this.SchoolYears.length) count++;
            if (this.LunchStatuses != null && this.LunchStatuses.length) count++;
            if (this.SpecialEducationStatuses != null && this.SpecialEducationStatuses.length) count++;
            if (this.Schools != null && this.Schools.length) count++;
            if (this.Districts != null && this.Districts.length) count++;

            return count;
        }

        isFiltering = () => {
            return this.filteringCount() > 0;
        }

        reset = () => {
            this.Grades = [];
            this.Ethnicities = [];
            this.SchoolYears = [];
            this.EnglishLanguageLearnerStatuses = [];
            this.LunchStatuses = [];
            this.SpecialEducationStatuses = [];
            this.SchoolYear = this.DefaultSchoolYear;
        }

        constructor(schoolYear?: number) {
            if (schoolYear) this.DefaultSchoolYear = schoolYear;
            this.reset();
        }
    }
}